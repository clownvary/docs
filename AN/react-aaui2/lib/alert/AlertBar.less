.alert-bar.show {
  z-index: 1;

  animation-name: fadeInDown;
}

.alert-bar {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;

  width: 100%;
  max-width: 600px;
  margin-right: auto;
  margin-left: auto;

  animation-name: fadeOutUp;
  animation-duration: 1s;

  animation-fill-mode: both;
}

// Thanks! @animate.css
@keyframes fadeInDown {
  from {
    transform: translate3d(0, -100%, 0);

    opacity: 0;
  }

  to {
    transform: none;

    opacity: 1;
  }
}

@keyframes fadeOutUp {
  from {
    opacity: 1;
  }

  to {
    transform: translate3d(0, -100%, 0);

    opacity: 0;
  }
}

.fadeOutUp {
  animation-name: fadeOutUp;
}
