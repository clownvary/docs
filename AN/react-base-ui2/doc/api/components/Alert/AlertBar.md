## AlertBar 



### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
timeout|number|&ensp;|false|&ensp;
type|enum:<br>['success', 'warning', 'danger', 'error', 'info']|&ensp;|false|&ensp;
onClose|func|noop|false|&ensp;
message|node|&ensp;|false|&ensp;
className|string|&ensp;|false|&ensp;