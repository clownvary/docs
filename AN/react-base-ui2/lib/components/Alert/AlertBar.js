'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _extends2 = require('babel-runtime/helpers/extends');var _extends3 = _interopRequireDefault(_extends2);var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);var _createClass2 = require('babel-runtime/helpers/createClass');var _createClass3 = _interopRequireDefault(_createClass2);var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);var _inherits2 = require('babel-runtime/helpers/inherits');var _inherits3 = _interopRequireDefault(_inherits2);var _noop = require('lodash/noop');var _noop2 = _interopRequireDefault(_noop);
var _omit = require('lodash/omit');var _omit2 = _interopRequireDefault(_omit);
var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _propTypes = require('prop-types');
var _classnames = require('classnames');var _classnames2 = _interopRequireDefault(_classnames);
var _Alert = require('./Alert');var _Alert2 = _interopRequireDefault(_Alert);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}var


AlertBar = function (_PureComponent) {(0, _inherits3.default)(AlertBar, _PureComponent);function AlertBar() {var _ref;var _temp, _this, _ret;(0, _classCallCheck3.default)(this, AlertBar);for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref = AlertBar.__proto__ || (0, _getPrototypeOf2.default)(AlertBar)).call.apply(_ref, [this].concat(args))), _this), _this.












    state = {
      show: false }, _this.

























    setRef = function (element) {
      _this.element = element;
    }, _this.

    handleClose = function () {
      _this.setState({
        show: false });


      _this.element &&
      _this.element.addEventListener('animationend', function () {
        _this.props.onClose();
      });
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);}(0, _createClass3.default)(AlertBar, [{ key: 'componentDidMount', value: function componentDidMount() {var _this2 = this;var timeout = this.props.timeout;this.animationFrame = setTimeout(function () {_this2.setState({ show: true });});if (timeout) {this.timer = setTimeout(function () {_this2.setState({ show: false });}, timeout);}} }, { key: 'componentWillUnmount', value: function componentWillUnmount() {clearTimeout(this.animationFrame);clearTimeout(this.timer);} }, { key: 'render', value: function render()

    {var _props =
      this.props,message = _props.message,type = _props.type,className = _props.className,rest = (0, _objectWithoutProperties3.default)(_props, ['message', 'type', 'className']);var
      show = this.state.show;
      var classes = (0, _classnames2.default)(
      {
        'alert-bar': true,
        show: show },

      className);


      return (
        _react2.default.createElement('div', { className: classes, ref: this.setRef },
          _react2.default.createElement(_Alert2.default, (0, _extends3.default)({},
            (0, _omit2.default)(rest, ['timeout']), {
              type: type,
              onClose: this.handleClose }),

            _react2.default.createElement('span', null, message))));



    } }]);return AlertBar;}(_react.PureComponent);AlertBar.propTypes = { timeout: _propTypes.number, type: (0, _propTypes.oneOf)(['success', 'warning', 'danger', 'error', 'info']), onClose: _propTypes.func, message: _propTypes.node, className: _propTypes.string };AlertBar.defaultProps = { onClose: _noop2.default };exports.default = AlertBar;module.exports = exports['default'];