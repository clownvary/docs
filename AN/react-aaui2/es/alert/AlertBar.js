import _extends from 'babel-runtime/helpers/extends';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import React, { PureComponent } from 'react';
import { number, string, func, node, oneOf } from 'prop-types';
import classNames from 'classnames';
import Alert from './Alert';

import { noop, omit } from '../shared/utils';
import './AlertBar.less';

var AlertBar = function (_PureComponent) {
  _inherits(AlertBar, _PureComponent);

  function AlertBar() {
    var _temp, _this, _ret;

    _classCallCheck(this, AlertBar);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, _PureComponent.call.apply(_PureComponent, [this].concat(args))), _this), _this.state = {
      show: false
    }, _this.setRef = function (element) {
      _this.element = element;
    }, _this.handleClose = function () {
      _this.setState({
        show: false
      });

      _this.element && _this.element.addEventListener('animationend', function () {
        _this.props.onClose();
      });
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  AlertBar.prototype.componentDidMount = function componentDidMount() {
    var _this2 = this;

    var timeout = this.props.timeout;


    this.animationFrame = setTimeout(function () {
      _this2.setState({
        show: true
      });
    });

    if (timeout) {
      this.timer = setTimeout(function () {
        _this2.setState({
          show: false
        });
      }, timeout);
    }
  };

  AlertBar.prototype.componentWillUnmount = function componentWillUnmount() {
    clearTimeout(this.animationFrame);
    clearTimeout(this.timer);
  };

  AlertBar.prototype.render = function render() {
    var _props = this.props,
        message = _props.message,
        type = _props.type,
        className = _props.className,
        rest = _objectWithoutProperties(_props, ['message', 'type', 'className']);

    var show = this.state.show;

    var classes = classNames({
      'alert-bar': true,
      show: show
    }, className);

    return React.createElement(
      'div',
      { className: classes, ref: this.setRef },
      React.createElement(
        Alert,
        _extends({}, omit(rest, ['timeout']), {
          type: type,
          onClose: this.handleClose
        }),
        React.createElement(
          'span',
          null,
          message
        )
      )
    );
  };

  return AlertBar;
}(PureComponent);

AlertBar.propTypes = {
  timeout: number,
  type: oneOf(['success', 'warning', 'danger', 'error', 'info']),
  onClose: func,
  message: node,
  className: string
};
AlertBar.defaultProps = {
  onClose: noop
};
export default AlertBar;