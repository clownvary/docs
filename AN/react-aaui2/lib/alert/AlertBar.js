'use strict';

exports.__esModule = true;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Alert = require('./Alert');

var _Alert2 = _interopRequireDefault(_Alert);

var _utils = require('../shared/utils');

require('./AlertBar.less');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var AlertBar = function (_PureComponent) {
  (0, _inherits3.default)(AlertBar, _PureComponent);

  function AlertBar() {
    var _temp, _this, _ret;

    (0, _classCallCheck3.default)(this, AlertBar);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, _PureComponent.call.apply(_PureComponent, [this].concat(args))), _this), _this.state = {
      show: false
    }, _this.setRef = function (element) {
      _this.element = element;
    }, _this.handleClose = function () {
      _this.setState({
        show: false
      });

      _this.element && _this.element.addEventListener('animationend', function () {
        _this.props.onClose();
      });
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
  }

  AlertBar.prototype.componentDidMount = function componentDidMount() {
    var _this2 = this;

    var timeout = this.props.timeout;


    this.animationFrame = setTimeout(function () {
      _this2.setState({
        show: true
      });
    });

    if (timeout) {
      this.timer = setTimeout(function () {
        _this2.setState({
          show: false
        });
      }, timeout);
    }
  };

  AlertBar.prototype.componentWillUnmount = function componentWillUnmount() {
    clearTimeout(this.animationFrame);
    clearTimeout(this.timer);
  };

  AlertBar.prototype.render = function render() {
    var _props = this.props,
        message = _props.message,
        type = _props.type,
        className = _props.className,
        rest = (0, _objectWithoutProperties3.default)(_props, ['message', 'type', 'className']);
    var show = this.state.show;

    var classes = (0, _classnames2.default)({
      'alert-bar': true,
      show: show
    }, className);

    return _react2.default.createElement(
      'div',
      { className: classes, ref: this.setRef },
      _react2.default.createElement(
        _Alert2.default,
        (0, _extends3.default)({}, (0, _utils.omit)(rest, ['timeout']), {
          type: type,
          onClose: this.handleClose
        }),
        _react2.default.createElement(
          'span',
          null,
          message
        )
      )
    );
  };

  return AlertBar;
}(_react.PureComponent);

AlertBar.propTypes = {
  timeout: _propTypes.number,
  type: (0, _propTypes.oneOf)(['success', 'warning', 'danger', 'error', 'info']),
  onClose: _propTypes.func,
  message: _propTypes.node,
  className: _propTypes.string
};
AlertBar.defaultProps = {
  onClose: _utils.noop
};
exports.default = AlertBar;
module.exports = exports['default'];