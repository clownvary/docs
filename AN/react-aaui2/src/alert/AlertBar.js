import React, { PureComponent } from 'react'
import { number, string, func, node, oneOf } from 'prop-types'
import classNames from 'classnames'
import Alert from './Alert'

import { noop, omit } from '../shared/utils'
import './AlertBar.less'

export default class AlertBar extends PureComponent {
  static propTypes = {
    timeout: number,
    type: oneOf(['success', 'warning', 'danger', 'error', 'info']),
    onClose: func,
    message: node,
    className: string,
  }

  static defaultProps = {
    onClose: noop,
  }

  state = {
    show: false,
  }

  componentDidMount() {
    const { timeout } = this.props

    this.animationFrame = setTimeout(() => {
      this.setState({
        show: true,
      })
    })

    if (timeout) {
      this.timer = setTimeout(() => {
        this.setState({
          show: false,
        })
      }, timeout)
    }
  }

  componentWillUnmount() {
    clearTimeout(this.animationFrame)
    clearTimeout(this.timer)
  }

  setRef = element => {
    this.element = element
  }

  handleClose = () => {
    this.setState({
      show: false,
    })

    this.element &&
      this.element.addEventListener('animationend', () => {
        this.props.onClose()
      })
  }

  render() {
    const { message, type, className, ...rest } = this.props
    const { show } = this.state
    const classes = classNames(
      {
        'alert-bar': true,
        show,
      },
      className,
    )

    return (
      <div className={classes} ref={this.setRef}>
        <Alert
          {...omit(rest, ['timeout'])}
          type={type}
          onClose={this.handleClose}
        >
          <span>{message}</span>
        </Alert>
      </div>
    )
  }
}
