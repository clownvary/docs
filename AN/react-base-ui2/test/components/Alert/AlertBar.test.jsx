import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import AlertBar from 'src/components/Alert/AlertBar';

beforeAll(() => {
  jest.useFakeTimers();
});

afterAll(() => {
  jest.useRealTimers();
});

const setup = (props = {}) => {
  const newProps = {
    onClose: () => {},
    ...props
  };

  const Component = <AlertBar {...newProps} />;
  const wrapper = mount(Component);

  return { Component, wrapper };
};

it('should render without errors', () => {
  const { wrapper } = setup();

  expect(wrapper).toBeTruthy();
});

it('should render right thing', () => {
  const { wrapper } = setup();

  expect(wrapper.hasClass('alert-bar')).toBe(true);
});

it('should render right thing and do nott change unexpected', () => {
  const { Component } = setup();
  const tree = renderer.create(Component).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should display Alert when setting `show` as true', () => {
  const { wrapper } = setup();

  wrapper.setState({ show: true });

  expect(wrapper.find('.alert-bar.show').length).toBe(1);
});

it('should hide Alert when clicking `Close` button', () => {
  const { wrapper } = setup();

  wrapper.setState({ show: true });
  expect(wrapper.state('show')).toBe(true);

  wrapper.find('button').simulate('click');
  expect(wrapper.state('show')).toBe(false);
});

it('should call `onClose` when closing Alert', () => {
  const { wrapper } = setup();

  wrapper.find('button').simulate('click');
});

it('should have right lifecycle', () => {
  const { wrapper } = setup();

  jest.runAllTimers();

  wrapper.instance().element.addEventListener = (_, fn) => fn();
  expect(wrapper.state().show).toBe(true);

  wrapper.find('button').simulate('click');
  expect(wrapper.state().show).toBe(false);

  wrapper.unmount();
});
