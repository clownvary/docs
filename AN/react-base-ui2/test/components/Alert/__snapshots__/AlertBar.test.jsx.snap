// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render right thing and do nott change unexpected 1`] = `
<div
  className="alert-bar"
>
  <div
    className="alert alert-info alert-dismissable"
    role="alert"
    style={undefined}
  >
    <span
      aria-label="info icon"
      className="icon icon-info-circle"
    />
    <span />
    <button
      className="close"
      onClick={[Function]}
      type="button"
    >
      Ã—
    </button>
  </div>
</div>
`;
